tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -949,7 +969,10 @@ void CommandQueue::enqueue_program(Program& program, bool blocking) {
     map<uint64_t, unique_ptr<Buffer>>& program_to_buffer = this->program_to_buffer(this->device->id());
     if (not program_to_buffer.count(program_id)) {
         stall = true;
-        ProgramMap program_to_device_map = ConstructProgramMap(this->device, program, this->dispatch_core);
+        uint8_t num_hw_cqs = this->device->num_hw_cqs();
+        tt_cxy_pair logical_dispatch_location = dispatch_core_manager::get(num_hw_cqs).command_dispatcher_core(this->device->id(), tt::Cluster::instance().get_assigned_channel_for_device(this->device->id()), this->command_queue_id);
+        CoreCoord logical_dispatch_core(logical_dispatch_location.x, logical_dispatch_location.y);
```

Comment:
Can this logic to determine `logical_dispatch_core` be pushed into `ConstructProgramMap`? This would lead to one less function parameter into `ConstructProgramMap`